### PR TITLE
Decompile miscellaneous functions

### PR DESCRIPTION
**Pull request by rr-**
#### Checklist

- [x] I have read the [coding conventions](https://github.com/LostArtefacts/TR1X/blob/master/CONTRIBUTING.md#coding-conventions)
- [x] I have added a changelog entry about what my pull request accomplishes, or it is an internal change

#### Description

#1693 

This PR decompiles a range of miscellaneous functions. After this process, we should only have specific object code remaining to decompile.

Since there's no straightforward pattern for testing, here are some key areas to consider:

- shooting mechanics (checking hits, misses, ricochet effects, Lara targeting enemies, enemies targeting Lara, and mercenaries targeting monks)
- gun flashes with various guns (quick check)
- flare lighting (quick check)
- carried items / item drops (especially enemies with multiple item drops)
- boat and skidoo driving mechanics (particularly bumping and acceleration)
- general collision detection with enemies and statics

### CODE REVIEW

**Comment by aredfan** [comment]
- [x] The monks/xian spearmen doesn't do damage to Lara.
20241218_180502_Barkhang_Monastery

- [x] Blood sprites appear in odd places when Lara is shooting at enemies.
20241218_180328_Catacombs_of_the_Talion

- [x] The boat has no collusion, Lara can swim through it.
20241218_180415_Venice

**Comment by rr-** [comment]
@aredfan fixed.

**Review by aredfan — APPROVED** [review]
> @aredfan fixed.

Thank you, LGTM. 👍

**Comment by lahm86** [comment]
I get a crash in Bartoli's below when the enemies shoot the windows. It looks like a div by 0 but it's interesting as this function has been decompiled for a while. I can't reproduce this on the develop branch.
```
libtrx/log_windows.c 58 M_StackTrace 00. 0x5fbeb228: (/app/build/tr2/win/game/los.c:363:39) LOS_CheckSmashable
libtrx/log_windows.c 46 M_StackTrace --- 0x5fbb0000: D:\Games\Dev\TR2X\TR2X.dll
libtrx/log_windows.c 58 M_StackTrace 01. 0x5fbc424d: (/app/build/tr2/win/game/creature.c:1027:35) Creature_ShootAtLara
```

image

File: src/tr2/game/creature.c [hunk 1]
```diff
@@ -952,3 +956,78 @@ bool Creature_IsAlly(const ITEM *const item)
...
+            .x = item->pos.x,
+            .y = item->pos.y - STEP_L * 3,
+            .z = item->pos.z,
+        },
+        .room_num = item->room_num,
+    };
+
+    GAME_VECTOR target = {
+        .pos = {
+            .x = item->pos.x,
```
Comment (lahm86) [inline src/tr2/game/creature.c]: Should this perhaps be `target_item` - could explain the window crash?

Comment (rr-) [reply src/tr2/game/creature.c]: @lahm86 thanks, that's a good catch, and you're absolutely right. Fixed.

**Review by lahm86 — APPROVED** [review]

**Comment by Richard-L** [comment]
@rr- I don't know if it's relevant, but in this build https://github.com/LostArtefacts/TRX/pull/2072#issuecomment-2551895812 Lara will not equip a weapon if you pick it in the inventory. It needs another equip key press for her to do so which is unexpected behaviour.

**Comment by aredfan** [comment]
> @rr- I don't know if it's relevant, but in this build [#2072 (comment)](https://github.com/LostArtefacts/TRX/pull/2072#issuecomment-2551895812) Lara will not equip a weapon if you pick it in the inventory. It needs another equip key press for her to do so which is unexpected behaviour.

That's an OG bug and it's captured here: https://github.com/LostArtefacts/TRX/issues/828

**Comment by Richard-L** [comment]
Ah thank you. Sorry still getting the lodown of all the existing issues now that TR2X is working for me.

**Pull request merged**